app/featuresmodel.h

```diff
@@ -120,6 +145,10 @@ class FeaturesModel : public QAbstractListModel
     FeatureLayerPairs mFeatures;
     QString mSearchExpression;
     QgsVectorLayer *mLayer = nullptr;
+
+    QAtomicInt mNextSearchId = 0;
+    QFutureWatcher<QgsFeatureList> mSearchResultWatcher;
+    bool mFetchingResults = false;
```

Comment:
Can we have this as a `Q_PROPERTY`? QML will then know when it changes between true/false and can, for example, show a spinner